shopfloor/actions/data.py

```diff
@@ -93,13 +93,19 @@ def package(self, record, picking=None, with_packaging=False, **kw):
         if with_packaging:
             parser += self._package_packaging_parser
         data = self._jsonify(record, parser, **kw)
+        qty = len(record.quant_ids)
         # handle special cases
         if data and picking:
-            lines = picking.move_line_ids.filtered(
-                lambda l: l.result_package_id == record
-                and l.state in ["partially_available", "assigned", "done"]
+            move_line_count = self.env["stock.move.line"].search_count(
+                [
+                    ("picking_id.picking_type_id", "=", picking.picking_type_id.id),
+                    ("result_package_id", "=", record.id),
+                    ("state", "in", ["partially_available", "assigned"]),
+                ]
             )
-            data.update({"move_line_count": len(lines)})
+            qty += move_line_count
+            # TODO does this name really makes sense?
+            data.update({"move_line_count": qty})
```

Comment:
As we are now mixing a number of quants + a number of move lines not `done`, this deserves a comment explaining why we are doing this.
I'm still trying to understand why the previous implementation was giving wrong results, it's not obvious.

Reply:
Because each processed qty is extracted in a new picking. So picking.move_line_ids.filtered(state="done") returns nothing. Moreover, there could also be other receptions on the same destination package.
The best result is given by counting what is already on the destination package.